readthedocs/api/v2/utils.py

```diff
@@ -121,6 +121,32 @@ def sync_versions_to_db(project, versions, type):  # pylint: disable=redefined-b
     return added
 
 
+def _create_versions_in_bulk(project, type, versions):
+    """
+    Create versions (tuple of version_id and version_name) in batch.
+
+    Returns the slug of all added versions.
+    """
+    added = set()
+    batch_size = 150
+    objs = (
+        Version(
+            project=project,
+            type=type,
+            identifier=version_id,
+            verbose_name=version_name,
+        )
+        for version_id, version_name in versions
+    )
+    while True:
+        batch = list(itertools.islice(objs, batch_size))
```

Comment:
I usually try to not use `while True` code in production. Can't we split `objs` in advance and iterate over them with a `for` loop instead? That's will be safer.

Reply:
splitting in advance will evaluate all objects at once making it slow.

Reply:
Well... We can split `versions` in advance instead, and create multiple `objs`

Reply:
python doesn't have something like `take n`, you'll be implementing the same with a while loop.

Reply:
This is the documented way of doing this in Django: https://docs.djangoproject.com/en/2.2/ref/models/querysets/#bulk-create